font/src/rusttype/mod.rs

```diff
@@ -72,6 +72,34 @@ impl ::Rasterize for RustTypeRasterizer {
     }
 
     fn get_glyph(&mut self, glyph_key: GlyphKey) -> Result<RasterizedGlyph, Error> {
+        match glyph_key.c {
+            super::UNDERLINE_CURSOR_CHAR => {
+                let metrics = self.metrics(glyph_key.font_key, glyph_key.size)?;
+
+                return super::get_underline_cursor_glyph(metrics.descent as i32, metrics.average_advance as i32);
+
+            },
+            super::BEAM_CURSOR_CHAR | super::BOX_CURSOR_CHAR => {
+                let metrics = self.metrics(glyph_key.font_key, glyph_key.size)?;
+
+                return if glyph_key.c == super::BEAM_CURSOR_CHAR {
```

Comment:
Is there a reason why you've added a nested `if` here which checks the cursor character again? If the goal is to remove the duplicate `metrics` call, the same should be done for the underline cursor.

I'd propose to put the `let metrics = ` before the match and then remove the nested `if` and replace it with another `match` branch.

Reply:
Yeah, will do. I was pretty much copy-pasting from the fontconfig code.